### PR TITLE
While validating VS subsets repeatings, consider hosts and subset combinations

### PR DESCRIPTION
**Pull request by hhovsepy**
Fix for https://github.com/kiali/kiali/issues/4458.

Create a VS without a subsets in route, like this:
```
      route:
        - destination:
            host: reviews
          weight: 80
        - destination:
            host: reviews.bookinfo2.svc.cluster.local
          weight: 10
        - destination:
            host: reviews.bookinfo.svc.cluster.local
          weight: 10
```

Before it was showing warning message in wrong place because validation was tightened to only subset:
Screenshot from 2021-11-23 10-02-14

Now it validates host+subset combination and considers host in fqdn and puts validation message on the host line:
Screenshot from 2021-11-23 09-06-35

### CODE REVIEW

**Comment by hhovsepy** [comment]
For QE, it needs to be tested with http, tcp and tls routes, also with and without subsets.

**Comment by hhovsepy** [comment]
Doc changes https://github.com/kiali/kiali.io/pull/489

File: models/istio_validation.go [hunk 1]
```diff
@@ -303,7 +303,7 @@ var checkDescriptors = map[string]IstioCheck{
 	},
 	"virtualservices.route.repeatedsubset": {
 		Code:     "KIA1105",
-		Message:  "This subset is already referenced in another route destination",
+		Message:  "This host subset combination is already referenced in another route destination",
```
Comment (lucasponce) [inline models/istio_validation.go:306]: Minor rewording, "host subset" -> "host/subset" ?

File: business/checkers/virtualservices/route_checker.go [hunk 2]
```diff
@@ -125,13 +128,13 @@ func (route RouteChecker) checkTlsRoutes() ([]*models.IstioCheck, bool) {
 			}
 		}
 
-		trackTcpTlsSubset(routeIdx, "http", destinationWeights, &validations)
+		route.trackTcpTlsSubset(routeIdx, "http", destinationWeights, &validations)
```
Comment (lucasponce) [inline business/checkers/virtualservices/route_checker.go:131]: Here the "http" -> "tls", not for this PR, but it was a bug from the refactoring to use the Istio client, on this case, it's checking the tls routes and the string is used to check collissions.

File: models/istio_validation.go (hunk 1 above)
Comment (hhovsepy) [reply models/istio_validation.go:306]: it is used 'host subset' also in "destinationrules.multimatch", might be changing both?

File: business/checkers/virtualservices/route_checker.go [hunk 3]
```diff
@@ -94,7 +97,7 @@ func (route RouteChecker) checkTcpRoutes() ([]*models.IstioCheck, bool) {
...
```
Comment (lucasponce) [inline business/checkers/virtualservices/route_checker.go:100]: Here the "http" -> "tcp" I think

File: models/istio_validation.go (hunk 1 above)
Comment (lucasponce) [reply models/istio_validation.go:306]: Nah, it's very minor, I'm also fine with the wording and if it's used in other places I don't see a I need to change.

**Review by lucasponce — APPROVED** [review]

**Review by skondkar — APPROVED** [review]
LGTM.

Validation with http route without subset:
Screenshot from 2021-12-01 18-21-24

No message if http route is with subset:
Screenshot from 2021-12-01 18-35-17

Also verified with tcp and tls routes, with and without subsets.

Screenshot from 2021-12-01 18-44-00

Screenshot from 2021-12-01 18-55-24

**Comment by skondkar** [comment]
Polarion test case >> SWSQE-2168

**Pull request merged**